### ISSUE TITLE
RIDE doesn't remember interpreter version information when connecting to a running interpreter

### ISSUE THREAD

**Issue by jayfoad** [Dyalog/ride#358]
### Describe the issue you are having
In RIDE 4.1 after connecting to a running interpreter, Help->About tells me that all the interpreter version fields are "unknown", as below. However in the protocol log I can see that the interpreter has sent all the required information:
```
2315 send ["Identify",{"identity":1}]
2315 send ["Connect",{"remoteId":2}]
2315 send ["GetWindowLayout",{}]
2561 recv SupportedProtocols=2
2561 recv UsingProtocol=2
2561 recv ["Identify",{"Port":0,"IPAddress":"","Vendor":"Dyalog Limited","Language":"APL","version":"17.0.32926","Machine":"jay-laptop","arch":"Unicode/64","Project":"CLEAR WS","Process":"","User":"jay","pid":19292,"token":"","date":"Created: Apr 25 2018 at 13:15:40","platform":"Linux-64"}]
```
This is a regression from RIDE 4.0.

### Paste the contents of Help → About (Shift+F1)
```
IDE:
  Version: 4.1.3171
  Platform: Linux x86_64
  Date: 2018-04-25 09:42:12 +0200
  Git commit: 6e614349fc3936835adfe25c5e59fccd5373424b
  Preferences:{
    "floatOnTop":"1",
    "floating":"1"
  }

Interpreter:
  Version: unknown
  Platform: unknown
  Edition: unknown
  Date: unknown
```

**Comment by e9gille** [comment]
I can't replicate this in RIDE from Win.
About from connection page shows unknown, but after connecting it does reflect correct versions

**Comment by jayfoad** [comment]
I can reproduce this sometimes, but as soon as I start debugging it the problem goes away :-(

**Comment by e9gille** [comment]
Fixed by c6f3b25
This issue would only manifest if a floating window had been created before about box opened. If floating windows is enabled, RIDE creates a hidden one on startup, so it would immediately cause this issue.
